api/provider/azure.go

```diff
@@ -38,7 +38,7 @@ func NewAzureProvider(ext conf.OAuthProviderConfiguration, scopes string) (OAuth
 	}
 
 	authHost := chooseHost(ext.URL, defaultAzureAuthBase)
-	apiPath := chooseHost(ext.URL, defaultAzureAPIBase)
+	apiPath := chooseHost("", defaultAzureAPIBase)
```

Comment:
Hmm this line actually breaks the test because we use a mock server to mock the responses returned by the azure endpoints. The test suite ends up making a call to the actual `graph.microsoft.com/oidc/userinfo` endpoint rather than the test server set up. 

The reason why this doesn't affect gitlab (which also allows you to add an external tenant url) is because the api host is the same for both oauth and user info. In Azure's case, it seems like they use `login.microsoftonline.com/{tenant_id}` for oauth calls and `graph.microsoft.com` for retrieving user info.